gutsy-gamblers/main.py

```diff
@@ -0,0 +1,256 @@
+        self.add_widget(self.now_marker)
+        self.add_widget(self.season_dial)
+
+        # workaround for kivy's LIFO draw order
+        if settings_modal is not None:
+            Clock.schedule_once(lambda dt: settings_modal.open(), 0.5)
+
+    def on_size(self, a, b):
+        # Maintains a constant aspect ratio of 0.5625 (16:9)
+        width, height = Window.size
+
+        if not Window.fullscreen and (height / width) != 0.5625:
+            height = width * 0.5625
+
+        width /= Metrics.density
+        height /= Metrics.density
+
+        Window.size = (width, height)
+
+    def time_control_button(self):
+        time_control_popup = TimeWizard(self.dial_widget, self)
+        time_control_popup.open()
+
+    def settings_button(self):
+        SettingsScreen().open()
+
+
+# Time control panel #
+class TimeWizard(Popup):
+    def __init__(self, dial, parent, **kwargs):
+        self.dial = dial
+        super(TimeWizard, self).__init__(**kwargs)
+        self.redraw_checkbox.bind(active=self.delta_override)
+
+        self.current_date.text = self.dial.date.strftime("%d/%m/%Y")
+        self.clock = Clock.schedule_interval(self.update_date, self.dial.midnight_delta)
+
+    def update_date(self, *args):
+        print('called update_date')
+        self.clock.cancel()
+        self.current_date.text = self.dial.date.strftime("%d/%m/%Y")
+        self.clock = Clock.schedule_interval(self.update_date, self.dial.midnight_delta)
+
+    def delta_override(self, *args):
+        print('called delta_override')
+        if self.redraw_checkbox.active is True:
+            self.dial.midnight_delta = 0.1
+            self.update_date()
+            self.dial.redraw()
+            # self.parent.speedy_time = True
+        else:
+            midnight = datetime.now() + timedelta(days=1)
+            # Good thing flake8 isn't 78 chars huh?
+            self.dial.midnight_delta = (datetime(year=midnight.year,
+                                                 month=midnight.month,
+                                                 day=midnight.day,
+                                                 hour=0,
+                                                 minute=0,
+                                                 second=0) - datetime.now()).seconds
+            self.dial.redraw()
+            # self.parent.speedy_time = False
+
+    def revert_date(self):
+        print('called revert_date')
+        self.dial.date = datetime.now()
+        self.update_date()
+        self.dial.redraw()
+
+
+# Settings panel #
+class UpdateLocationButton(Button):
+    """
+    Button for updating location.
+    """
+    settingsscreen = ObjectProperty()
+
+    latlon = ConfigParserProperty(
+        '', 'global', datahelpers.LOCATION_LATLON, 'app', val_type=str)
+    friendlyname = ConfigParserProperty(
+        '', 'global', datahelpers.LOCATION_FRIENDLY, 'app', val_type=str)
+
+    latlong_regex = re.compile(r'^(\-?\d+(\.\d+)?),\s*(\-?\d+(\.\d+)?)$')
```

Comment:
complicated regular expressions should be broken up into multiple lines with inline comments explaining what the different parts do for improved readability.